docs/how-to-guides/meson-args.rst

```diff
@@ -17,5 +16,98 @@ Passing arguments to Meson_
    - Mention the order in which Meson arguments are added, and how it affect
      Meson
 
+Advanced meson options can be accessed by specifying extra arguments to the individual meson comands:
+
+   - meson dist: https://mesonbuild.com/Commands.html#dist
+   - meson setup: https://mesonbuild.com/Commands.html#setup
+   - meson compile: https://mesonbuild.com/Commands.html#compile
+   - meson install: https://mesonbuild.com/Commands.html#install
+
+These arguments can be added to the project via the project's ``pyproject.toml``:
+
+.. code-block:: console
+
+   [tool.meson-python.args]
+   dist = ['dist-argument_1','dist-argument_2', '...']]
+   setup = ['setup-argument_1','setup-argument_2', '...']]
+   compile = ['compile-argument_1','compile-argument_2', '...']]
+   install = ['install-argument_1','install-argument_2', '...']]
+
+.. admonition:: All four categories must exist
+   :class: attention
+
+   Currently all four option categories (dist, setup, compile and install) must be present in the [tool.meson-python.args] section, even if no arguments are supplied (empty list: [])
```

Comment:
Please wrap key names in ``. Also, I think the brackets in `tool.meson-python.args` can be omitted.

---

> Currently all four option categories [...] must be present

Uh, this is a major oversight, that should definitely not be the case :sweat_smile:

Reply:
thought so actually, but it didn't bother me enough to fix it :-D
... I created an issue, (https://github.com/mesonbuild/meson-python/issues/293)
If have time in the future I will see if I can make a PR forthis

Reply:
> Also, I think the brackets in `tool.meson-python.args` can be omitted.

I get: ``ERROR Failed to parse ./pyproject.toml: Expected '=' after a key in a key/value pair (at line 91, column 23) `` if I remove the brackets arrount ``tool.meson-python.args``?

Reply:
> Please wrap key names in ``

You mean like this? I haven't seen this before (admittedly I haven't seem many pyproject.toml files yet). Is there a reason to use the " wrapping instead of just writing the key?